WordPress/src/main/java/org/wordpress/android/ui/plugins/PluginDetailActivity.java

```diff
@@ -167,15 +216,44 @@ public void onSitePluginUpdated(OnSitePluginUpdated event) {
                     + event.error.message);
             return;
         }
-        if (event.plugin != null) {
-            mPlugin = event.plugin;
-            refreshStates();
+        mPlugin = mPluginStore.getSitePluginByName(mSite, mPlugin.getName());
+        refreshViews();
+    }
+
+    @SuppressWarnings("unused")
+    @Subscribe(threadMode = ThreadMode.MAIN)
+    public void onPluginInfoChanged(PluginStore.OnPluginInfoChanged event) {
+        if (event.isError()) {
+            AppLog.e(AppLog.T.API, "An error occurred while fetching the plugin info with type: "
+                    + event.error.type);
+            return;
+        }
+        if (event.pluginInfo != null && mPlugin.getSlug().equals(event.pluginInfo.getSlug())) {
```

Comment:
Should there be a check whether `mPlugin.getSlug()` is null here?

Reply:
I am a bit thorn about this. `Slug` should definitely be there for a plugin, because without that, we couldn't actually get the plugin information from the wp.org plugin repository on the server side. And if there were to be an NPE, it'd happen here instead: `mDispatcher.dispatch(PluginActionBuilder.newFetchPluginInfoAction(mPlugin.getSlug()));`.

Having said that, I've seen some weird behavior before with plugin endpoints which we had to fix server side. I am inclined to leave this as is and let it crash if there is an edge case. When we check the crash, If we find that `slug` field could be empty for a valid plugin, we'll add the check, otherwise, we'll fix this on the server side. What do you think?